storage/storage.go

```diff
@@ -30,7 +40,7 @@ type Backend interface {
 	PutFile(ctx context.Context, url string, path string) error
 	// Determines whether this backends supports the given request (url/path/class).
 	// A backend normally uses this to match the url prefix (e.g. "s3://")
-	Supports(url string) error
+	Supports(url string, class tes.FileType, op Operation) error
```

Comment:
This seems fine, but we could also try supporting a put.

Reply:
I cant think of a use case where there is going to be a server accepting anonymous POST requests.

Reply:
One case would be an http endpoint inside a private network, but it doesn't always need to be anonymous.